bentoml/_internal/resource.py

```diff
@@ -211,25 +214,29 @@ def query_os_cpu_count() -> int:
 #         raise ValueError(f"Invalid MEM resource limit '{mem}'")
 
 
-class NvidiaGpuResource(Resource[int], resource_id="nvidia.com/gpu"):
+class NvidiaGpuResource(Resource[t.List[int]], resource_id="nvidia.com/gpu"):
     @classmethod
-    def from_spec(cls, spec: t.Any) -> int:
-        if not isinstance(spec, (int, str)):
-            raise TypeError("NVidia GPU resource limit must be int or str")
-
-        if isinstance(spec, int):
-            return spec
+    def from_spec(cls, spec: t.Union[int, str, t.List[int | str]]) -> t.List[int]:
+        if not isinstance(spec, (int, str, t.List)):
+            raise TypeError(
+                "NVidia GPU resource limit must be int, str or a list specifing the exact GPUs to use."
+            )
 
         try:
-            return int(spec)
+            if isinstance(spec, int):
+                return list(range(spec))
+            elif isinstance(spec, str):
+                return list(range(int(spec)))
+            else:
+                return [int(x) for x in spec]
         except ValueError:
             raise BentoMLConfigException(
                 f"Invalid NVidia GPU resource limit '{spec}'. "
             )
 
     @classmethod
     @functools.lru_cache(maxsize=1)
-    def from_system(cls) -> int:
+    def from_system(cls) -> t.List[int]:
```

Comment:
```suggestion
    def from_system(cls) -> list[int]:
```

Reply:
thanks for the review @aarnphm 🙂! can you tell me the difference between the two types signatures here?

Reply:
See [PEP604](https://peps.python.org/pep-0604/)

Reply:
ahh got it. thanks!